micro_sam/sam_3d_wrapper.py

```diff
@@ -6,16 +6,32 @@
 from segment_anything.modeling.image_encoder import window_partition, window_unpartition
 from segment_anything.modeling import Sam
 
-from .util import get_sam_model
-
-
-def get_3d_sam_model(device, n_classes, image_size, model_type="vit_b"):
-    predictor, sam = get_sam_model(
-        return_sam=True, model_type=model_type, device=device, num_multimask_outputs=n_classes,
-        flexible_load_checkpoint=True, image_size=image_size,
+from .util import get_sam_model, _load_checkpoint, _handle_checkpoint_loading
+
+
+def get_3d_sam_model(
+    device,
+    n_classes,
+    image_size,
+    model_type="vit_b",
+    checkpoint_path=None,
+):
+    _, sam = get_sam_model(
```

Comment:
Can't we just pass the checkpoint here? Then we don't need to duplicate the code below.

Reply:
Ah, that won't work as we need to pass the weights to the wrapper model (to initialize the adapter blocks)

Reply:
Hmm, I don't fully understand. Let's discuss tomorrow :) .

Reply:
Okay now I see already where your suspicion comes from.

TLDR: I think we might have to merge the `get_3d_sam_model` into `get_sam_model` for the best possible design (which leads to flexibly loading SAM checkpoints for finetuning, and the 3d-SAM checkpoints for downstream semantic inference).

I have a plan on this. I'll take care of this first thing in the morning. Thanks for spotting.

Reply:
I think the current design should work at least for intialization. We can revisit this later to discuss how we do this for actually loading the trained 3d models.

Reply:
Okie, I'll leave it to here for now then. Thanks!